class-zoninator-permissions.php

```diff
@@ -0,0 +1,51 @@
+<?php
+
+
+class Zoninator_Permissions
+{
+    public function check( $action = '', $zone_id = null ) {
+        // TODO: should check if zone locked
+        switch( $action ) {
+            case 'insert':
+                $verify_function = 'current_user_can_add_zones';
+                break;
+            case 'update':
+            case 'delete':
+                $verify_function = 'current_user_can_edit_zones';
+                break;
+            default:
+                $verify_function = 'current_user_can_manage_zones';
+                break;
+        }
+
+        if( ! call_user_func( array( $this, $verify_function ), $zone_id ) ) {
```

Comment:
Can we just call the method directly within the `switch` rather than using `call_user_func`? Feels like that would make it easier to read/follow.